haystack/nodes/prompt/prompt_node.py

```diff
@@ -0,0 +1,822 @@
+            raise ValueError(f"Expected prompt params {self.prompt_params} but got {list(available_params)}")
+
+        template_dict["prompt_template"] = self.prompt_text
+        return template_dict
+
+
+PREDEFINED_PROMPT_TEMPLATES = [
+    PromptTemplate(
+        name="question-answering",
+        prompt_text="Given the context please answer the question. Context: $documents; Question: $questions; Answer:",
+        prompt_params=["documents", "questions"],
+    ),
+    PromptTemplate(
+        name="question-generation",
+        prompt_text="Given the context please generate a question. Context: $documents; Question:",
+        prompt_params=["documents"],
+    ),
+    PromptTemplate(
+        name="conditioned-question-generation",
+        prompt_text="Please come up with a question for the given context and the answer. "
+        "Context: $documents; Answer: $answers; Question:",
+        prompt_params=["documents", "answers"],
+    ),
+    PromptTemplate(
+        name="summarization", prompt_text="Summarize this document: $documents Summary:", prompt_params=["documents"]
+    ),
+    PromptTemplate(
+        name="question-answering-check",
+        prompt_text="Does the following context contain the answer to the question. "
+        "Context: $documents; Question: $questions; Please answer yes or no! Answer:",
+        prompt_params=["documents", "questions"],
+    ),
+    PromptTemplate(
+        name="sentiment-analysis",
+        prompt_text="Please give a sentiment for this context. Answer with positive, "
+        "negative or neutral. Context: $documents; Answer:",
+        prompt_params=["documents"],
+    ),
+    PromptTemplate(
+        name="multiple-choice-question-answering",
+        prompt_text="Question:$questions ; Choose the most suitable option to answer the above question. "
+        "Options: $options; Answer:",
+        prompt_params=["questions", "options"],
+    ),
+    PromptTemplate(
+        name="topic-classification",
+        prompt_text="Categories: $options; What category best describes: $documents; Answer:",
+        prompt_params=["documents", "options"],
+    ),
+    PromptTemplate(
+        name="language-detection",
+        prompt_text="Detect the language in the following context and answer with the "
+        "name of the language. Context: $documents; Answer:",
+    ),
+    PromptTemplate(
+        name="translation",
+        prompt_text="Translate the following context to $target_language. Context: $documents; Translation:",
+    ),
+]
+
+
+class PromptModelInvocationLayer:
+    """
+    PromptModelInvocationLayer implementations execute a prompt on an underlying model.
+
+    The implementation can be a simple invocation on the underlying model running in a local runtime, or
+    could be even remote, for example, a call to a remote API endpoint.
+    """
+
+    def __init__(self, model_name_or_path: str, max_length: Optional[int] = 100, **kwargs):
```

Comment:
If the max_length value is optional I think the default value should be None. What do you think?

If we need the default value to be set then I would recommend removing the Optional type.